aioredis/commands/server.py

```diff
@@ -112,9 +125,21 @@ def flushdb(self):
         fut = self._conn.execute('FLUSHDB')
         return wait_ok(fut)
 
-    def info(self, section):
-        """Get information and statistics about the server."""
-        # TODO: check section
+    def info(self, section='default'):
+        """Get information and statistics about the server.
+
+        If called without argument will return default set of sections.
+        For available sections, see http://redis.io/commands/INFO
+
+        :raises TypeError: if section is not string
+        :raises ValueError: if section is invalid
+
+        """
+        if not isinstance(section, str):
+            raise TypeError("section must be str")
+        section = section.lower()
+        if section not in self.INFO_ARGS:
+            raise ValueError("invalid section")
```

Comment:
Good point about forward compatibility. I included the strict sections validation because I thought it would be more helpful to tell you exactly what was the issue when someone provides a wrong section.

I don't think there is any validation that the args should be a string done at `RedisConnection.execute()` and `util.encode_command()` just checks that it's one of bytes, str, int or float. Unless I missed something.

Right now, I just get some StopIteration exceptions if I provide an invalid section. Unless I'm doing something wrong.

Here is the example I used:

```python
loop = asyncio.get_event_loop()

async def go():
    redis = await aioredis.create_redis(('localhost', 32769), loop=loop)
    val = await redis.info(5)  # or redis.info('whatever')
    print(val)
    redis.close()

loop.run_until_complete(go())
```

And the error I get from the above:

```
Traceback (most recent call last):
  File "/Users/thanos/Dev/forks/aioredis/aioredis/util.py", line 205, in wait_convert
    return type_(result)
  File "/Users/thanos/Dev/forks/aioredis/aioredis/commands/server.py", line 229, in parse_info
    section = next(block)[2:].lower()
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "thanos.py", line 12, in <module>
    loop.run_until_complete(go())
  File "/Users/thanos/.pyenv/versions/3.5.2/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/Users/thanos/.pyenv/versions/3.5.2/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Users/thanos/.pyenv/versions/3.5.2/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "thanos.py", line 8, in go
    val = await redis.info('')
RuntimeError: generator raised StopIteration
Task was destroyed but it is pending!
task: <Task pending coro=<RedisConnection._read_data() running at /Users/thanos/Dev/forks/aioredis/aioredis/connection.py:131> wait_for=<Future pending cb=[Task._wakeup()]> cb=[Future.set_result()]>
```

Or I could just update the PR and keep the default arg only if you think it's the best way to go.

Either way, thanks for your time to review it :)